ydb/library/yql/core/cbo/cbo_optimizer_new.h

```diff
@@ -74,6 +79,54 @@ enum EJoinKind: ui32
 EJoinKind ConvertToJoinKind(const TString& joinString);
 TString ConvertToJoinString(const EJoinKind kind);
 
+/**
+ * This is a temporary structure for KQP provider
+ * We will soon be supporting multiple providers and we will need to design
+ * some interfaces to pass provider-specific context to the optimizer
+*/
+struct IProviderContext {
+    virtual ~IProviderContext() = default;
+
+    virtual double ComputeJoinCost(const TOptimizerStatistics& leftStats, const TOptimizerStatistics& rightStats, EJoinImplType joinImpl) const = 0;
+
+    virtual bool IsJoinApplicable(const std::shared_ptr<IBaseOptimizerNode>& left, 
+        const std::shared_ptr<IBaseOptimizerNode>& right, 
+        const std::set<std::pair<NDq::TJoinColumn, NDq::TJoinColumn>>& joinConditions,
+        EJoinImplType joinImpl) = 0;
+
+};
+
+/**
+ * Temporary solution for default provider context
+*/
+
+struct TDummyProviderContext : IProviderContext {
+    TDummyProviderContext() {}
+
+    double ComputeJoinCost(const TOptimizerStatistics& leftStats, const TOptimizerStatistics& rightStats, EJoinImplType joinImpl) const {
+        Y_UNUSED(joinImpl);
+        return leftStats.Nrows + 2.0 * rightStats.Nrows;
+    }
+
+    bool IsJoinApplicable(const std::shared_ptr<IBaseOptimizerNode>& left, 
+        const std::shared_ptr<IBaseOptimizerNode>& right, 
+        const std::set<std::pair<NDq::TJoinColumn, NDq::TJoinColumn>>& joinConditions,
+        EJoinImplType joinImpl) {
+
+        Y_UNUSED(left);
+        Y_UNUSED(right);
+        Y_UNUSED(joinConditions);
+        Y_UNUSED(joinImpl);
+
+        return true;
+    }
+
+    static const TDummyProviderContext& instance() {
```

Comment:
Why do we need singleton here?

Reply:
I added it here to avoid to create a new DummyProvider every time.